app/src/main/java/org/piwigo/helper/NetworkHelper.java

```diff
@@ -12,13 +12,30 @@ public NetworkHelper() {
         INSTANCE = this;
     }
 
-    public boolean hasInternet(Context context) {
+    public NetworkInfo getCurrentNetwork(Context context)
+    {
         ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
-        NetworkInfo currentNetwork = cm.getActiveNetworkInfo();
+        NetworkInfo network = cm.getActiveNetworkInfo();
+
+        return (network);
+    }
+
+    public boolean hasInternet(Context context) {
+        NetworkInfo currentNetwork = getCurrentNetwork(context);
 
         if (currentNetwork != null) {
             return (currentNetwork.isConnected());
         }
         return (false);
     }
+
+    public int getNetworkType(Context context)
+    {
+        NetworkInfo currentNetwork = getCurrentNetwork(context);
+
+        if (currentNetwork != null) {
+            return (currentNetwork.getType());
+        }
+        return (0);
```

Comment:
Is there no symbolic name for 0?
Also we should add a comment for this function explaining the meaning of the result values. - But as the return value of "currentNetwork.getType()" might change, it would be bad do copy-paste the description, but giving a reference to [NetworkInfo.getType()](https://developer.android.com/reference/android/net/NetworkInfo.html#getType())  in the comment would be great.
And: If you follow this link you'll see, that the used method is already deprecated. So if possible it would be great to directly check the alternative solution.

Reply:
See https://github.com/Piwigo/Piwigo-Android/pull/214/commits/da7580e1d5695d60237c385fd427e1c05c70c53c